frontend/src/metabase/query_builder/components/view/QuestionDataSource.unit.spec.js

```diff
@@ -305,100 +262,95 @@ jest.mock("metabase/core/components/Link", () => ({ to: href, ...props }) => (
 describe("QuestionDataSource", () => {
   const GUI_TEST_CASE = {
     SAVED_GUI_QUESTION: {
-      question: getSavedGUIQuestion(),
+      card: { ...BASE_GUI_QUESTION, ...SAVED_QUESTION },
       questionType: "saved GUI question",
     },
     AD_HOC_QUESTION: {
-      question: getAdHocQuestion(),
+      card: BASE_GUI_QUESTION,
       questionType: "ad-hoc GUI question",
     },
     SAVED_GUI_PRODUCTS_JOIN: {
-      question: getSavedGUIQuestion({
+      card: {
+        ...BASE_GUI_QUESTION,
+        ...SAVED_QUESTION,
         dataset_query: QUERY_WITH_PRODUCTS_JOIN,
-      }),
+      },
       questionType: "saved GUI question joining a table",
     },
     AD_HOC_PRODUCTS_JOIN: {
-      question: getAdHocQuestion({ dataset_query: QUERY_WITH_PRODUCTS_JOIN }),
+      card: { ...BASE_GUI_QUESTION, dataset_query: QUERY_WITH_PRODUCTS_JOIN },
```

Comment:
This line shows how we avoid one unnecessary jump with this new change. What we really want to do here is construct the card data that we'll feed later to the question.

Before the change I found myself constantly switching context by jumping to different function calls, just to figure out what are they doing.